bats/tests/k8s/helm-install-rancher.bats

```diff
@@ -13,12 +13,95 @@ local_setup() {
     helm repo update
 }
 
+# Check that the rancher-latest/rancher helm chart at the given version is
+# supported on the current Kubernetes version (as determined by
+# $RD_KUBERNETES_PREV_VERSION)
+assert_rancher_chart_compatible() {
+    output="" # Hack: tell shellcheck about this variable
+    local chart_version=$1
+
+    run helm show chart rancher-latest/rancher --version "$chart_version"
+    assert_success || return
+
+    run awk '/^kubeVersion:/ { $1 = ""; print }' <<<"$output"
+    assert_success || return
+    # We only support kubeVersion of form "< x.y.z"
+    assert_output --regexp '^[[:space:]]*<[[:space:]]*[^[:space:]]+$' || return
+
+    run awk '{ print $2 }' <<<"$output"
+    assert_success || return
+
+    local unsupported_version=$output
+    semver_gt "$unsupported_version" "$RD_KUBERNETES_PREV_VERSION" || return
+}
+
+# Set (and save) $rancher_chart_version to $RD_RANCHER_IMAGE_TAG if it is set
+# (and compatible), or otherwise the oldest chart version that supports
+# $RD_KUBERNETES_PREV_VERSION.
+# If no compatible chart version could be found, calls mark_k3s_version_skipped
+# and fails the test.
+determine_chart_version() {
+    local rancher_chart_version
+    if [[ -n $RD_RANCHER_IMAGE_TAG ]]; then
+        # If a version is given, check that it's compatible.
+        rancher_chart_version=${RD_RANCHER_IMAGE_TAG#v}
+        if ! assert_rancher_chart_compatible "$rancher_chart_version"; then
+            mark_k3s_version_skipped
+            printf "Rancher %s is not compatible with Kubernetes %s" \
+                "$rancher_chart_version" "$RD_KUBERNETES_PREV_VERSION" |
+                fail
+            return
+        fi
+        save_var rancher_chart_version
+        return
+    fi
+    local default_version
+    default_version=$(rancher_image_tag)
+    default_version=${default_version#v}
+
+    run --separate-stderr helm search repo --versions rancher-latest/rancher --output json
+    assert_success || return
+
+    run jq_output 'map(.version).[]'
+    assert_success || return
+
+    run sort --version-sort <<<"$output"
+    assert_success || return
+    local versions=$output
+
+    for rancher_chart_version in $versions; do
```

Comment:
There should be another `local rancher_chart_version` here to match the code above.